### DISCUSSION TITLE
<iwxxm:remarks> field in MetarSpeci

### DISCUSSION THREAD

**Discussion by Mattonius5** [wmo-im/iwxxm#292] [Q&A] [answered]
Hello everyone,
When we convert METAR from TAC to IWXXM, RMK field (from TAC) is lost in translation.
We got a request for keeping RMK field in IWXXM for Metars, Speci and others.
When checking official XSD schemas, it seems we discard that field correctly as there is no remarks field in [https://schemas.wmo.int/iwxxm/3.0/metarSpeci.xsd](url) 
There are only three Opmet types where "<iwxxm:remarks>" element is allowed, see below:
[https://schemas.wmo.int/iwxxm/3.0/](http://example.com/)
spaceWxAdvisory.xsd
tropicalCycloneAdvis.xsd
volcanicAshAdvisory.xsd

My question:
Should the RMK field stay in IWXXM Metar? If yes, then what field should it be mapped to?
What does the official specification say for this case?

Thank you for answers,
Matt

**Comment by jkorosi** [comment, marked as answer]
Hi Matt,

The national extension, coded in the RMK section, may be placed to <extension/> elements. For this purpose, you must define an XSD specification to create a valid IWXXM report. You can inspire yourself by @mgoberfield guidance **Encoding supplementary information in IWXXM** available at https://github.com/NOAA-MDL/GIFTs/wiki/Encoding-supplementary-information-in-IWXXM.

Best regards,
Jan

**Reply by Mattonius5** [reply]
Thank you, Jan, for your reply and provided link to the article. 
Can you check whether do I understand this correctly?
1) If I change the XSD, other providers, who read our IWXXM messages, must update their XSD to allow for the possibility that the field arrives; otherwise, validation against the schema will cause an error.
2) Only the national agency is authorized to propose and specify such changes. The article keep mentioning "State" and "State extensions"). 
Best regards,
Matt

**Reply by jkorosi** [reply]
1. Yes, all consumers must update the national XSD schemas used in <extension/> element. Without that, the validation may fail.
2. I suppose so, but not sure if this is somehow regulated. @mgoberfield, @blchoy and @dzinkhan are more qualified to respond to this question.

**Reply by Mattonius5** [reply]
Thank you for confirming.

[IWXXM guidelines summaries this also quite well.](https://www.icao.int/APAC/Documents/edocs/IWXXM%20Guidelines%20v4.pdf)
**3.1.2. Variances to IWXXM**
Where a State has a requirement to exchange additional elements (such as remark sections) within their
IWXXM, the State will be required to design, test and implement an extension to the IWXXM schema. These
national extensions can only be supported when implemented in a globally standardized way. The use of
extensions within the IWXXM is discouraged and should only be utilized where absolutely necessary.

**Reply by mgoberfield** [reply]
Many thanks to @jkorosi for his responses while I was on leave.

Hello @Mattonius5,

My comments on your points as follows: 1) The desired way to add additional content to IWXXM products is to create a separate schema file that is specific to your country's needs/requirements, not to copy the WMO IWXXM schema files and modify those; 2) Yes, the schema files you will create needs your country's regulatory agency input and approval.  It would be the same agency that regulates/governs the content of your country's METAR, SPECI and other products.
